lib/improver/cube_combiner.py

```diff
@@ -47,72 +47,274 @@ class CubeCombiner(object):
+
+        metatype = 'DimCoord'
+        if 'metatype' in changes:
+            if changes['metatype'] == 'AuxCoord':
+                new_coord_method = iris.coords.AuxCoord
+                metatype = 'AuxCoord'
+            else:
+                new_coord_method = iris.coords.DimCoord
+        else:
+            new_coord_method = iris.coords.DimCoord
+        result = cube
+        points = changes['points']
+        bounds = None
+        if 'bounds' in changes:
+            bounds = changes['bounds']
+        units = None
+        if 'units' in changes:
+            units = changes['units']
+        new_coord = new_coord_method(long_name=coord_name,
+                                     points=points,
+                                     bounds=bounds,
+                                     units=units)
+        result.add_aux_coord(new_coord)
+        if metatype == 'DimCoord':
+            result = iris.util.new_axis(result, coord_name)
+        if self.warnings_on:
+            msg = ("Adding new coordinate "
+                   "{} with {}".format(coord_name,
+                                       changes))
+            warnings.warn(msg)
+        return result
+
+    def update_coord(self, cube, coord_name, changes):
+        """Amend the metadata in the combined cube.
+
+        Args:
+            cube (iris.cube.Cube):
+                Cube containing combined data.
+            coord_name (string):
+                Name of the coordinate being updated.
+            changes (string or dict):
+                Details on coordinate to be updated.
+                If changes = 'delete' the coordinate is deleted.
+
+        Returns:
+            result (iris.cube.Cube):
+                Cube with updated coordinate.
+
+        Raises:
+            ValueError : Can only remove a coordinate of length 1
+            ValueError : Mismatch in points in existing coord
+                and updated metadata.
+            ValueError : Mismatch in bounds in existing coord
+                and updated metadata.
+            ValueError : The shape of the bounds array should
+                be points.shape + (n_bounds,)
+            UserWarning: Deleted coordinate.
+            UserWarning: Updated coordinate
+
+        """
+        new_coord = cube.coord(coord_name)
+        result = cube
+        if changes == 'delete':
+            if len(new_coord.points) != 1:
+                msg = ("Can only remove a coordinate of length 1"
+                       " coord  = {}".format(coord_name))
+                raise ValueError(msg)
+            result.remove_coord(coord_name)
+            result = iris.util.squeeze(result)
+            if self.warnings_on:
+                msg = ("Deleted coordinate "
+                       "{}".format(coord_name))
+                warnings.warn(msg)
+        else:
+            if 'points' in changes:
+                new_points = np.array(changes['points'])
+                if (len(new_points) ==
```

Comment:
I think that this if/else block can be reduced to just `new_coord.points = new_points`, as Iris will already generate an exception if someone tries to assign an array of the wrong size to the points in a coordinate, and the exception of `ValueError: New points shape must match existing points shape.` is pretty descriptive of the problem.

Reply:
But it will not tell you which coordinate had the problem.  Are you sure it is worth making this change?